crates/host-bench/src/lib.rs

```diff
@@ -89,19 +87,62 @@ pub struct HostArgs {
     pub input_path: Option<PathBuf>,
 }
 
+/// Segments based on total trace cells across all chips
+#[derive(Debug, Clone, Copy)]
+pub struct TraceSizeSegmentationStrategy {
+    max_height: usize,
+    max_cells: usize,
+}
+
+impl TraceSizeSegmentationStrategy {
+    pub fn new(max_height: usize, max_cells: usize) -> Self {
+        assert!(max_height < (1 << 27));
+        Self { max_height, max_cells }
+    }
+}
+
+impl SegmentationStrategy for TraceSizeSegmentationStrategy {
+    fn should_segment(
+        &self,
+        air_names: &[String],
+        trace_heights: &[usize],
+        trace_cells: &[usize],
+    ) -> bool {
+        for (i, &height) in trace_heights.iter().enumerate() {
+            if height > self.max_height {
+                tracing::info!(
+                    "Should segment because chip {i} (name: {}) has height {height}",
+                    air_names[i],
+                );
```

Comment:
I think with these stricter strategies we have it makes sense to add this

```suggestion
                tracing::info!(
                    "Should segment because chip {i} (name: {}) has height {height} > {}",
                    air_names[i],
                    self.max_height
                );
```